src/sqlfluff/core/templaters/placeholder.py

```diff
@@ -29,8 +29,10 @@
     "pyformat": regex.compile(
         r"(?<![:\w\x5c])%\((?P<param_name>[\w_]+)\)s", regex.UNICODE
     ),
-    # e.g. WHERE bla = $name
-    "dollar": regex.compile(r"(?<![:\w\x5c])\$(?P<param_name>[\w_]+)", regex.UNICODE),
+    # e.g. WHERE bla = $name or WHERE bla = ${name}
+    "dollar": regex.compile(
```

Comment:
is it a potential problem that this would accept asymmetric braces? `${foo`

Reply:
Fair. But same in `ampersand` that does this. I think it's low risk and not worth the more complicated Regex so ensure this doesn't happen. Garbage in, garbage out!

Reply:
Had a look and,  because of the named group, changing it to an or doesn't work. So would need a new `dollar_braces` group, which would be a breaking change if did same for `ampersand`.

So still think it's best to leave as is.

Reply:
Makes sense - esp if that's how it works in others